### PR TITLE
Nav bar changes

### PR DESCRIPTION
**Pull request by surtich**

### CODE REVIEW

**Comment by yamalight** [comment]
@surtich sorry, but this is still too much in one PR - can you please further split it?
I'd prefer to have small (1-3 commits) PRs that focus on one thing. 
E.g. in this PR I'd leave only commits: 4db3867, 80e73cb, eb28854, 7203587 (so, related to navbar itself)

**Comment by surtich** [comment]
@yamalight, I am sorry. I knew I had to do smaller pull request but it has been impossible.

The reason is that I am using this project in my classes and the [exam](https://github.com/DAWZayas/building-products-with-js/tree/experts-enunciado-primera) I also do with the project and I could not publish the changes until the students have done the exam.

I think it would be easier if you checked each file that changed as a whole from the first commit of the pull request to the last one.

You can see the names of the changed files with this instruction from my [fork](https://github.com/DAWZayas/building-products-with-js):

```bash
git diff cdf23e04eaf25dec996a738b40004b063514098a --name-only
```

An then to see the changes in each file: 
```bash
git diff cdf23e04eaf25dec996a738b40004b063514098a <path to file name>
```

**Comment by yamalight** [comment]
@surtich the main reason I want to have all those changes as separate PRs is - I already see a number of things I want to discuss in-depth (e.g. the reason to have additional `redirect.html` file, using OAuth on client, etc). This would make tracking changes in this PR very hard to follow.

**Comment by surtich** [comment]
@yamalight I understand the problem that you raise but I do not know how to do what you ask. At first I tried to make commits as orderly as possible. But it is inevitable that when implementing new features discover errors or improvements of previous things. If I extract commits that are not consecutive in the log history to make a PR of each functionality, they will produce multiple conflicts difficult to solve.

**Comment by yamalight** [comment]
@surtich I think creating new branches for each change and cherry-picking needed commits to them should work fine in this case, no?

**Comment by surtich** [comment]
@yamalight Cherry picking will probably resolve the conflicts but I see two aditional problems:

* This is a lot of work because I would have to do a cherry picking per file and not by commit.

* The worst thing is that the code generated in each branch is not going to be able to run and test because it will lack the changes in the files of the commits not included. As it stands right now, every commit is incremental and functionally runnable.

I think you could go commit to commit by making comments as you had started and if you see that something has been fixed later, you can delete the comment or simply ignore it because I already will have noticed.

**Comment by yamalight** [comment]
@surtich another point on why it's important to make commits that do only one thing 😛 
I'll re-open the [initial PR](https://github.com/yamalight/building-products-with-js/pull/21) and will go through it then. But it'll probably take a few days.

**Comment by surtich** [comment]
@yamalight  Do not worry, take your time :stuck_out_tongue:

I agree to make a unique commit by functionality and in fact is what I tried to do: Commit every time I have something runnable. That is why there are so many commits (35 for 8 functionalities).

But things like the following are inevitable:

- There are functionalities that are interdependent.
- Find out later that something you've implemented is not working properly. I know the best thing in this case is to rebase and in fact I have done it a couple of times, but it is not always so easy to go through the whole log history.

I think by reviewing the log history, you will see that in most cases every commit has a why.
